packages/flutter/lib/src/material/text_field.dart

```diff
@@ -92,7 +92,11 @@ class _TextFieldSelectionGestureDetectorBuilder extends TextSelectionGestureDete
       switch (Theme.of(_state.context).platform) {
         case TargetPlatform.iOS:
         case TargetPlatform.macOS:
-          renderEditable.selectWordEdge(cause: SelectionChangedCause.tap);
```

Comment:
This is weird. I thought whether the text position handling is position- or word-based should be completely decided by the input device, i.e. Android or Windows touch devices should also select words based on word, while macOS should almost always be based on position. (Not only to this function but also `onSingleLongTapMoveUpdate` and `onSingleLongTapStart`)

Also AFAIK the `TargetPlatform` is defined more like how it looks instead of where it runs. +cc @stuartmorgan

Reply:
FWIW, I tested on an iPad with the Apple keyboard. Tapping words selects to word edges. Using the trackpad instead selects by position.

Reply:
Yeah I think we should ditch target platform and decide solely based on the pointer kind. Perhaps even have a global function that decides whether the kind is a precision device.

Reply:
On Android, doing a touch tap does a precise placement of the cursor. The word edge behavior is a Mac/iOS/iPadOs thing. So this should remain platform-dependent.

Reply:
My point was in support of this current implementation. I don't think it'll be avoidable to have to account for both dimensions since native behaviors are so diverse.

Reply:
Ok, this 2-dimensional criteria looks good to me.